### PR TITLE
improve ratelimit doc

### PR DESCRIPTION
**Pull request by zirain**
Please provide a description for what this PR is for.

- add test
- improve global ratelimit EnvoyFilter

fix: https://github.com/istio/istio.io/issues/10372

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [x] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

**Comment by zirain** [comment]
cc @douglas-reid

**Comment by douglas-reid** [comment]
@kyessenov can you review this?  it seems good at first pass, but i'd like a more rate-limit savvy set of eyes to confirm.

**Comment by ericvn** [comment]
@kyessenov Any thoughts? Thanks.

**Review by kyessenov — APPROVED** [review]
Thanks, this is great, definitely improving the doc.

File: content/en/docs/tasks/policy-enforcement/rate-limit/test.sh [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+# apply local ratelimit envoyfilter
+snip_local_rate_limit_2
+
+# wait a while for envoyfilter
+sleep 1s
+
+for i in {1..10}; do
+    echo "$i"
+    snip_verify_local_rate_limit_1
+done
```
Comment (ericvn) [inline content/en/docs/tasks/policy-enforcement/rate-limit/test.sh:60]: Is this just extra debugging? The verify_same should repeat the snip_verify_local_rate_limit_1 and compare output until the condition is met or the retries have expired.

File: content/en/docs/tasks/policy-enforcement/rate-limit/test.sh [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+_set_ingress_environment_variables
+export GATEWAY_URL=$INGRESS_HOST:$INGRESS_PORT
+echo "*** GATEWAY_URL = $GATEWAY_URL ***"
+
+# apply global ratelimit envoyfilter
+snip_global_rate_limit_3
+snip_global_rate_limit_4
+
+# verify global ratelimit
+snip_verify_global_rate_limit_1
```
Comment (ericvn) [inline content/en/docs/tasks/policy-enforcement/rate-limit/test.sh:47]: I don't think this line is needed, as it's done as part of the next line.

File: content/en/docs/tasks/policy-enforcement/rate-limit/test.sh (hunk 1 above)
Comment (zirain) [reply content/en/docs/tasks/policy-enforcement/rate-limit/test.sh:60]: you're right, I misunderstand the usage of `_verify_same`.
will fix this, thanks!

File: content/en/docs/tasks/policy-enforcement/rate-limit/test.sh (hunk 2 above)
Comment (zirain) [reply content/en/docs/tasks/policy-enforcement/rate-limit/test.sh]: updated

**Review by ericvn — APPROVED** [review]

**Pull request merged**